packages/trader/src/AppV2/Components/TradeParameters/Duration/container.tsx

```diff
@@ -0,0 +1,126 @@
+import React, { useEffect, useState } from 'react';
+import { ActionSheet, CaptionText, Text } from '@deriv-com/quill-ui';
+import { Localize } from '@deriv/translations';
+import { useTraderStore } from 'Stores/useTraderStores';
+import DurationEndTimePicker from './datepicker';
+import { observer } from '@deriv/stores';
+import DurationChips from './chips';
+import DurationWheelPicker from './wheelpicker';
+
+const DurationActionSheetContainer = observer(
+    ({ selected_hour, setSelectedHour }: { selected_hour: number[]; setSelectedHour: (arg: number[]) => void }) => {
+        const { duration, duration_unit, duration_units_list, onChangeMultiple, expiry_time } = useTraderStore();
+        const [unit, setUnit] = useState(expiry_time ? 'et' : duration_unit);
+        const [selected_time, setSelectedTime] = useState([duration]);
+        const [expiry_date_data, setExpiryDate] = useState<Date>(new Date());
+        const [end_time, setEndTime] = useState<string>('');
+        const [toggle_picker, setTogglePicker] = useState<boolean>(false);
+
+        const onChangeUnit = (value: string) => {
+            if (unit !== 'h') {
+                setSelectedHour([]);
+            }
+            setUnit(value);
+        };
+
+        const handleSelectExpiryDate = (date: Date) => {
+            setExpiryDate(date);
+            const current_date = new Date();
+            const timeDifference = +date - +current_date;
+            const dayDifference = Math.ceil(timeDifference / (1000 * 60 * 60 * 24));
+            setSelectedTime([dayDifference]);
+        };
+
+        useEffect(() => {
+            if (duration_unit === 'm' && duration > 59) {
+                const hour = Math.floor(duration / 60);
+                const minutes = duration % 60;
+                setSelectedHour([hour, minutes]);
+                setUnit('h');
+            }
+        }, [duration, duration_unit, setSelectedHour]);
+
+        const setWheelPickerValue = (index: number, value: string | number) => {
+            if (unit === 'h') {
+                const arr = selected_hour;
+                arr[index] = Number(value);
+                setSelectedHour(arr);
+            } else if (unit == 'd') {
+                setSelectedTime([Number(value)]);
+                const updated_date = new Date();
+                updated_date.setDate(updated_date.getDate() + Number(value));
+                setExpiryDate(updated_date);
+            } else {
+                setSelectedTime([Number(value)]);
+            }
+        };
+
+        return (
+            <div className='duration-container'>
+                <ActionSheet.Header title={<Localize i18n_default_text='Duration' />} />
+                <DurationChips duration_units_list={duration_units_list} onChangeUnit={onChangeUnit} unit={unit} />
```

Comment:
how is this working without wrapping inside ActionSheet.body

Reply:
i only see header and footer

Reply:
its in parent